internal/collector/collector.go

```diff
@@ -113,8 +115,20 @@ func (k *KeepalivedCollector) Collect(ch chan<- prometheus.Metric) {
 
 	keepalivedUp := float64(1)
 
-	keepalivedStats, err := k.getKeepalivedStats()
-	if err != nil {
+	b := backoff.NewExponentialBackOff()
+	b.InitialInterval = 10 * time.Millisecond
+	b.MaxElapsedTime = 2 * time.Second
+	b.Reset()
+
+	var keepalivedStats *KeepalivedStats
+
+	if err := backoff.Retry(func() error {
+		var err error
+		keepalivedStats, err = k.getKeepalivedStats()
+		logrus.WithError(err).Debug("Failed to get keepalived stats. Retrying...")
```

Comment:
I think we should surround it with `err != nil` cause if case the err is nil then we shouldn't log the failure.

Reply:
Damn it! :)))